cr.h

```diff
@@ -1138,15 +1168,27 @@ static int cr_plugin_main(cr_plugin &ctx, cr_op operation) {
     auto p = (cr_internal *)ctx.p;
 #ifndef __MINGW32__
     __try {
-#endif
         if (p->main) {
             return p->main(&ctx, operation);
         }
-#ifndef __MINGW32__
     } __except (cr_seh_filter(ctx, GetExceptionCode())) {
         return -1;
     }
+#else
+    if (int sig = __builtin_setjmp(env)) {
+      ctx.version = ctx.last_working_version;
+      ctx.failure = cr_signal_to_failure(sig);
+      CR_LOG("1 FAILURE: %d (CR: %d)\n", sig, ctx.failure);
+      return -1;
+    } else {
+      auto p = (cr_internal *)ctx.p;
```

Comment:
You're shadowing the p from the start of the function here, I think you can remove this line

Reply:
@fungos  the line was removed, well spotted